bugwarrior/config/schema.py

```diff
@@ -172,98 +163,72 @@ def compute_data(cls, values):
 
-    log_level: typing.Literal[
-        ('DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL', 'DISABLED')
-    ] = 'INFO'
-    log_file: typing.Optional[LoggingPath] = None
+    log_level: Literal["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL", "DISABLED"] = (
+        "INFO"
+    )
+    log_file: typing.Optional[ExpandedPath] = None
 
 
-class Hooks(pydantic.v1.BaseModel):
-    pre_import: ConfigList = ConfigList([])
+class Hooks(BaseConfig):
+    pre_import: ConfigList = []
 
 
-class Notifications(pydantic.v1.BaseModel):
+class Notifications(BaseConfig):
     notifications: bool = False
-    # Although upstream supports it, pydantic has problems with Literal[None].
-    backend: typing.Optional[
-        typing.Literal[('gobject', 'growlnotify', 'applescript')]
-    ] = None
+    backend: typing.Optional[Literal["gobject", "growlnotify", "applescript"]] = None
     finished_querying_sticky: bool = True
     task_crud_sticky: bool = True
     only_on_new_tasks: bool = False
 
 
-class SchemaBase(pydantic.v1.BaseSettings):
-    class Config(PydanticConfig):
-        # Allow extra top-level sections so all targets don't have to be selected.
-        extra = 'ignore'
-
-    hooks: Hooks = Hooks()
-    notifications: Notifications = Notifications()
-
+class SchemaBase(BaseConfig):
+    # Allow extra top-level sections so all targets don't have to be selected.
+    model_config = ConfigDict(extra="ignore")
 
-class ValidationErrorEnhancedMessages(list):
-    """Methods loosely adapted from pydantic.error_wrappers."""
+    hooks: Hooks = Field(default_factory=Hooks)
+    notifications: Notifications = Field(default_factory=Notifications)
 
-    def __init__(self, error: pydantic.v1.ValidationError):
-        super().__init__(self.flatten(error))
 
-    def __str__(self):
-        return '\n'.join(self)
+def get_validation_error_enhanced_messages(
+    error: pydantic.ValidationError,
+) -> list[str]:
+    errors = []
+    for _error in error.errors():
+        msg = _error["msg"]
+        if _error["type"] == "extra_forbidden":
+            msg = "unrecognized option"
+        loc = _error["loc"]
+        loc_len = len(loc)
 
-    @staticmethod
-    def display_error_loc(e):
-        loc_len = len(e['loc'])
-        if loc_len == 1 or e['loc'][1] == '__root__':
-            return f"[{e['loc'][0]}]"
+        if loc_len == 1 or (loc_len > 1 and loc[1] == "__root__"):
+            formatted_error_loc = f"[{loc[0]}]"
         elif loc_len == 2:
-            # TODO We should be able to display the value itself:
-            # https://github.com/samuelcolvin/pydantic/issues/784
-            return f"[{e['loc'][0]}]\n{e['loc'][1]}"
-        raise ValueError(
-            'Configuration should not be nested more than two layers deep.'
-        )
+            formatted_error_loc = f"[{loc[0]}]\n{loc[1]}"
+            if _error["type"] not in {"missing", "extra_forbidden"}:
+                formatted_error_loc = f"{formatted_error_loc} = '{_error['input']}'"
```

Comment:
IMO it should look as similar as possible to the configuration file, which will never have single quotes. This looks particularly bad with arrays and tables which end up having nested single quotes.

```suggestion
                formatted_error_loc = f"{formatted_error_loc} = {_error['input']}"
```